Integration-cli: change daemon id generation from time to rand

tophj-ibm opened the pull request:
Fixes #24805

Changes test daemon id generation from using time.Now().UnixNano() to using
crypto rand.Read(). This is because of a go bug (fixed upstream for 1.8)
on ppc64le and s390x where time.Now().UnixNano() would not actually
return nanosecond precision, so initializing consecutive daemons sometimes
would result in them having the same id.

Signed-off-by: Christopher Jones tophj@linux.vnet.ibm.com

ping @tonistiigi 

![time dog](http://images.buycostumes.com/mgen/merchandiser/31693.jpg)

tonistiigi commented on integration-cli/daemon.go:
```diff
@@ -61,7 +63,11 @@ func NewDaemon(c *check.C) *Daemon {
+	randomBytes := make([]byte, 8)
+	_, err = rand.Read(randomBytes)
+	c.Assert(err, checker.IsNil, check.Commentf("could not read bytes in daemon id generation"))
+	id := fmt.Sprintf("d%d", binary.BigEndian.Uint64(randomBytes)%100000000)
```
nit: just `hex.EncodeToString()` would be simpler here. I don't think there is any reason this should be a number. It's even bad atm when finding the logs because the IDs have different lengths.

cpuguy83 replied:
Why not use stringid?

tophj-ibm replied:
Didn't know about stringId, but that looks like it should do exactly what I want

tophj-ibm committed:
Integration-cli: change daemon id generation

Fixes #24805

Changes test daemon id generation from using time.Now().UnixNano() to using
stringid. This is because of a go bug (fixed upstream for 1.8) on ppc64le
and s390x where time.Now().UnixNano() would not return nanosecond precision,
so initializing consecutive daemons sometimes would result in them having
the same id.

Signed-off-by: Christopher Jones <tophj@linux.vnet.ibm.com>

tophj-ibm commented:
updated to use the shortID from stringid. The ID is now 13 characters long as opposed to 9 characters long, but I didn't see where that should matter.

tonistiigi commented:
LGTM

cpuguy83 commented:
LGTM
